staging/src/k8s.io/apiextensions-apiserver/pkg/apihelpers/helpers.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+Copyright 2020 Authors of Arktos - file modified.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// File modified by cherrypick from kubernetes on 02/25/2021
+package apihelpers
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	"k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+)
+
+// IsProtectedCommunityGroup returns whether or not a group specified for a CRD is protected for the community and needs
+// to have the v1beta1.KubeAPIApprovalAnnotation set.
+func IsProtectedCommunityGroup(group string) bool {
+	switch {
```

Comment:
Wonder whether we should add a case for "arktos.futurewei.com"? We have already defined a CRD of our own in pkg/controller/artifacts/crd-network.yaml. We may have more in the future.

Reply:
checked bunch of information for this protectedcommunitygroup, This is only for an API in *.k8s.io or *.kubernetes.io is or is not approved.  For CRDs outside those groups, this doesn't applied.  For CRDs inside those groups, The *.k8s.io or *.kubernetes.io groups are owned by the Kubernetes community and protected by API review (see What APIs need to be reviewed.
	// See https://github.com/kubernetes/enhancements/pull/1111 for more details.